app/components/ITunesCard/index.js

```diff
@@ -0,0 +1,96 @@
+/**
+ *
+ * ITunesCard
+ *
+ */
+
+import React, { useRef } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import 'antd/dist/antd.css';
+import { Card, Popover } from 'antd';
+import { PlayCircleOutlined, PauseCircleOutlined } from '@ant-design/icons';
+import T from '@components/T';
+import { colors } from '@app/themes/index';
+
+const CustomCard = styled(Card)`
+  && {
+    margin: 1.25em 0;
+    box-shadow: ${colors.boxShadow};
+  }
+`;
+const Wrapper = styled.div`
+  display: flex;
+  alig-items: center;
+  justify-content: center;
+  margin-bottom: 0.75rem;
+  gap: 1em;
+`;
+const CustomButton = styled.button`
+  displaty: flex;
+  align-items: center;
+  justify-content: center;
+  background-color: ${colors.buttonBg};
+  border: none;
+  border-radius: 0.2em;
+  cursor: pointer;
+`;
+const CustomImg = styled.img`
+  display: block;
+  width: 45%;
+  margin: 0.75rem auto;
+`;
+
+export function ITunesCard({ track, onActionClick }) {
+  const audioRef = useRef(null);
+  const { artistName, trackName, collectionName, trackPrice, primaryGenreName, previewUrl, artworkUrl100 } = track;
+
+  const playerAction = {
+    play: 'PLAY',
+    pause: 'PAUSE'
+  };
+
+  const handleTrackPlay = (action) => {
+    if (action === playerAction.play) {
+      audioRef.current.play();
+      onActionClick(audioRef);
```

Comment:
so in this particular use case, we only care if play is pressed but this isn't generic. 

onActionClick always needs to be propagated up. 

You can check if the audio is playing or not in the container and then pause based on that.

Reply:
Okay. will try to refactor and handle the logic in iTunes container